test/parallel/test-aborted-util.js

```diff
@@ -0,0 +1,44 @@
+// Flags: --expose-gc
+'use strict';
+
+const common = require('../common');
+const { aborted } = require('util');
+const assert = require('assert');
+const { getEventListeners } = require('events');
+
+{
+  // Test aborted works when provided a resource
+  const ac = new AbortController();
+  aborted(ac.signal, {}).then(common.mustCall());
+  ac.abort();
+  assert.strictEqual(ac.signal.aborted, true);
+  assert.strictEqual(getEventListeners(ac.signal, 'abort').length, 0);
+}
+
+{
+  // Test aborted with gc cleanup
+  const ac = new AbortController();
+  aborted(ac.signal, {}).then(common.mustNotCall());
+  setImmediate(() => {
+    global.gc();
+    ac.abort();
+    assert.strictEqual(ac.signal.aborted, true);
+    assert.strictEqual(getEventListeners(ac.signal, 'abort').length, 0);
+  });
+}
+
+{
+  // Fails with error if not provided abort signal
+  const sig = new EventTarget();
+  assert.rejects(aborted(sig, {}), {
+    name: 'TypeError',
+  });
+}
+
+{
+  // Fails if not provided a resource
+  const ac = new AbortController();
+  assert.rejects(aborted(ac.signal, null), {
+    name: 'TypeError',
+  });
+}
```

Comment:
Can we have a test with top-level await as well?
```suggestion
}

{
  const childProcess = spawn(process.execPath, ['--input-type=module']);
  childProcess.on('exit', common.mustCall((code) => {
    assert.strictEqual(code, 13);
  });
  childProcess.stdin.end(`
    import { aborted } from 'node:util';
    await aborted(new AbortController().signal, {});
  `);
}
```

Reply:
I think yes can do, testing it